cmd/account/mgmt/account-unassign.go

```diff
@@ -0,0 +1,405 @@
+	ops := newAccountUnassignOptions(streams, flags)
+	accountUnassignCmd := &cobra.Command{
+		Use:               "unassign",
+		Short:             "Unassign account to user",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.printFlags.AddFlags(accountUnassignCmd)
+	accountUnassignCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountUnassignCmd.Flags().StringVarP(&ops.username, "username", "u", "", "LDAP username")
+	accountUnassignCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "Account ID")
+
+	return accountUnassignCmd
+}
+
+type accountUnassignOptions struct {
+	awsClient    awsprovider.Client
+	username     string
+	payerAccount string
+	accountID    string
+
+	flags      *genericclioptions.ConfigFlags
+	printFlags *printer.PrintFlags
+	genericclioptions.IOStreams
+}
+
+func newAccountUnassignOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *accountUnassignOptions {
+	return &accountUnassignOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+func (o *accountUnassignOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+	if o.username == "" && o.accountID == "" {
+		return cmdutil.UsageErrorf(cmd, "Please provide either an username or account ID")
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) run() error {
+
+	var (
+		accountUsername string
+		accountIdList   []string
+		destinationOU   string
+		rootID          string
+	)
+
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	if o.payerAccount == "osd-staging-1" {
+		rootID = OSDStaging1RootID
+		destinationOU = OSDStaging1OuID
+	} else if o.payerAccount == "osd-staging-2" {
+		rootID = OSDStaging2RootID
+		destinationOU = OSDStaging2OuID
+	} else {
+		return fmt.Errorf("invalid payer account provided")
+	}
+
+	o.awsClient = awsClient
+
+	if o.accountID != "" {
+
+		err := o.checkForHiveNameTag(o.accountID)
+		if err != nil {
+			return err
+		}
+		accountIdList = append(accountIdList, o.accountID)
+
+	}
+
+	if o.username != "" {
+		// Check that username is not a hive
+		if strings.HasPrefix(o.username, "hive") {
+			return ErrHiveNameProvided
+		}
+
+		accountUsername = o.username
+		accountIdList, err = o.listAccountsFromUser(accountUsername)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("Are you sure you want to unassign the accounts? [y/n] ")
+	reader := bufio.NewReader(os.Stdin)
+
+	response, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+
+	if response == "y" || response == "Y" {
```

Comment:
This isn't working for me during testing, because the var `response` includes the delimiter set on ReadString, so in actuality the `response` is `y\n` or `Y\n`.

Also, since this is an easy exit condition, we should be able to negate this check and exit early here, rather than put all of the logic inside a large `if` block, like this:

```go
response = strings.ToLower(response[0:1]) #only get the first letter of the string, and always lowercase it
if response != "y" {
    os.Exit(0)
}

for _, id := range accountIdList {
...
```

Reply:
By only grabbing the first character and lowercasing it, we allow users to be able to do things like `yes | osdctl account mgmt unassign` for future scriptablity without having to do any changes on our end, and then we don't have to check _every_ capitalization permutation of Yes:

```
yes
yeS
yEs
yES
Yes
YeS
YEs
YES
```